### ISSUE TITLE
3.18.1 - Issue with Connecting in General

### ISSUE THREAD

**Issue by falcon-usa** [NordSecurity/nordvpn-linux#424]
This update on Arch Linux (Endevour OS) returns an error message of "The VPN connection has failed. Please check your internet connection and try connecting to the VPN again. If the issue persists, contact our customer support."

**Comment by bsteiner36** [comment]
I am having the same issue in Redhat on 8.10 if I upgrade nordvpn to 3.18.1 I can no longer connect.  I had to revert my VM back to 3.17.2 in order to get nordvpn to work.  I might build out a new vm and try it on a fresh install to see if it still fails.

**Comment by mariusSincovici** [comment]
There is a new release 3.18.2 available.
Could you check if this fixes your issues?

**Comment by bsteiner36** [comment]
3.18.2. Did not resolve my issues. I am still failing to connect to nordvpn.  I am running without issues on 3.17.2.

**Comment by mariusSincovici** [comment]
Would it be possible to check in the system logs if there is a problem related to DNS configuration? 
Search in logs after `Failed to set DNS configuration` and if you find it pass the full error message.

**Comment by bsteiner36** [comment]
Here are some of the journalctl errors that I am seeing when attempting to connect.  
I also removed /var/lib/sss/db folder and restarted sssd since it mentioned there were issues with sssd when I did the dnf upgrade on NordVPN.

Jun 13 21:08:45 hostname nordvpnd[819]: 2024-06-14T01:08:45.126186Z ERROR libmoose::tracker::tracker: ContextNotFound Error while fetching shared context: ContextNotFound
Jun 13 21:08:46 hostname nordvpnd[819]: 2024/06/13 21:08:46 [Error] invoking dbus object method: Unit systemd-resolved.service not loaded.
Jun 13 21:08:46 hostname nordvpnd[819]: 2024/06/13 21:08:46 [Error] networker setting dns: setting dns with resolvconf: setting dns with resolvconf: Failed to set DNS configuration: Unit dbus-org.freedesktop.re
solve1.service not found.: exit status 1
Jun 13 21:08:46 hostname nordvpnd[819]: 2024/06/13 21:08:46 [Error] err1: autoconnect failure | err2: <nil>

**Comment by mariusSincovici** [comment]
Thanks for checking the logs.
Yes, this is the issue that we are working on to fix for the next release.
Until them please use an older version, 3.17.2.

**Comment by bsteiner36** [comment]
Thank you.  Please let me know if there is a new version to test.  I have a separate vm for testing this out so I can confirm it works before putting it into my production.

**Comment by bitterskittles** [comment]
3.18.3 did not fix this issue for me.

**Comment by mariusSincovici** [comment]
@bsteiner36 there is a new version 3.18.3, could you check if it fixed the problem? And if is still reproducing, could you send us the error message?

**Comment by mariusSincovici** [comment]
Hi @bitterskittles,

Could you please give is some more info about the system that you're using and if possible send the error message reported by the app from the system logs?
Thanks

**Comment by bsteiner36** [comment]
> @bsteiner36 there is a new version 3.18.3, could you check if it fixed the problem? And if is still reproducing, could you send us the error from the system logs?

I will take a look at this sometime today.  I will let you know if I still have issues and I will upload any logs I get.

**Comment by mariusSincovici** [comment]
Thanks @bitterskittles ,
Could you please check the system logs for the daemon, e.g. `journalctl -f -u snap.nordvpn.nordvpnd.service`, and see the error root cause?

**Comment by bitterskittles** [comment]
@mariusSincovici 

Thank you. Here are the most recent logs:
```
> journalctl -f -u snap.nordvpn.nordvpnd.service
Jul 24 08:35:51 localhost.localdomain nordvpn.nordvpnd[1699]: 2024/07/24 08:35:51 [Info] TELIO(v4.3.3): "telio::device::wg_controller":176 Removing peer: Some(Peer { public_key: "K53l...DlE=", endpoint: Some(94.46.222.35:51820), ip_addresses: [100.64.0.1, 10.5.0.1], persistent_keepalive_interval: Some(25), allowed_ips: [V4(Ipv4Network { addr: 0.0.0.0, prefix: 0 })], rx_bytes: Some(92), tx_bytes: Some(180), time_since_last_handshake: Some(758.771727ms), preshared_key: None })
Jul 24 08:35:51 localhost.localdomain nordvpn.nordvpnd[1699]: 2024/07/24 08:35:51 [Info] TELIO(v4.3.3): "telio::ffi":444 telio_stop entry with instance id: 11970393624792703573.
Jul 24 08:35:51 localhost.localdomain nordvpn.nordvpnd[1699]: task stopped - Telio
Jul 24 08:35:51 localhost.localdomain nordvpn.nordvpnd[1699]: task stopped - Wg
Jul 24 08:35:51 localhost.localdomain nordvpn.nordvpnd[1699]: 2024/07/24 08:35:51 [Info]{"type":"node","body":{"identifier":"3de51383-a991-4107-9ec2-79ec181e57ea","public_key":"***","nickname":null,"state":"disconnected","link_state":null,"is_exit":true,"is_vpn":true,"ip_addresses":["10.5.0.1","100.64.0.1"],"allowed_ips":["0.0.0.0/0"],"endpoint":"94.46.222.35:51820","hostname":null,"allow_incoming_connections":false,"allow_peer_send_files":false,"path":"direct"}}
Jul 24 08:35:51 localhost.localdomain nordvpn.nordvpnd[1699]: task stopped - Nurse
Jul 24 08:35:51 localhost.localdomain nordvpn.nordvpnd[1699]: task stopped - Nurse Heartbeat Analytics
Jul 24 08:35:51 localhost.localdomain nordvpn.nordvpnd[1699]: task stopped - Nurse QoS Analytics
Jul 24 08:35:52 localhost.localdomain nordvpn.nordvpnd[1699]: 2024/07/24 08:35:52 [Info] unset allow list
Jul 24 08:35:52 localhost.localdomain nordvpn.nordvpnd[1699]: 2024/07/24 08:35:52 [Error] networker setting dns: dns not set, no dns setting method is available
```

**Comment by mariusSincovici** [comment]
Thanks this is useful.
The app is not able to set the DNS configuration in the OS.
Do you have some custom DNS configuration in the OS and what are you using for DNS resolver in the OS?

**Comment by bitterskittles** [comment]
I am using NetworkManager which comes enabled by default in OpenSUSE Tumbleweed.
My default network connection is Wi-Fi to my home router.
I defined custom DNS on this connection because my broadband provider blocks certain websites through DNS, which includes nordvpn.com, torproject.org etc
![Wi-Fi config IPv4](https://github.com/user-attachments/assets/fadc65c0-4e69-4d5e-bff6-edcdc9bfc1cc)
![Wi-Fi config IPv6](https://github.com/user-attachments/assets/fa6551eb-f046-4709-9ebd-a981458b7058)


Here is a more comprehensive log file that includes the logs from other services:
[nordvpn-linux-logs.txt](https://github.com/user-attachments/files/16358932/nordvpn-linux-logs.txt)

**Comment by mariusSincovici** [comment]
Great info, thanks.
We've registered a bug on our side and we'll try to fix it in the next releases.
Until then would it be ok for you to use the rpm package instead of snap?

**Comment by bitterskittles** [comment]
@mariusSincovici 
Thank you for looking into the issue and the recommending to use the RPM package.

I uninstalled the Snap package and installed v3.18.3 using the installation script referenced here:
https://support.nordvpn.com/hc/en-us/articles/20196094470929-Installing-NordVPN-on-Linux-distributions

It worked out of the box using the default settings.

**Comment by bsteiner36** [comment]
So I am able to connect to nordvpn servers however I am unable to log into the server that is running nordvpn when it is active.  I have whitelisted my subnets but still can't get to the server.  I can ping out from my server.  This seems to be different than what it was like before.  I was able to get to the server using ssh previously with nordvpn running.

**Comment by mariusSincovici** [comment]
@bsteiner36 Could it be that you're affected by the other issue https://github.com/NordSecurity/nordvpn-linux/issues/512?

**Comment by bsteiner36** [comment]
> @mariusSincovici Thank you for looking into the issue and the recommending to use the RPM package.
> 
> I uninstalled the Snap package and installed v3.18.3 using the installation script referenced here: https://support.nordvpn.com/hc/en-us/articles/20196094470929-Installing-NordVPN-on-Linux-distributions
> 
> It worked out of the box using the default settings.

Yes this is exactly the same issue I am having. I have resources on my server running nord and i can not access them with the new version.  I can ping them but no ports are accessible including ssh.

**Comment by sokol-usa** [comment]
I am using Endeavor OS (Arch OS based) (updated as of 7.24.2024) and have recently updated to the latest NordVPN version 3.18.3. I am happy to report that using the install via the Arch User Repository (AUR) and I am having no connectivity issues at this moment.

**Comment by mariusSincovici** [comment]
I'm closing this because there are too many topics discussed, and the original issue seams to be fixed.

We have registered on our board the bug with DNS and snap: https://github.com/NordSecurity/nordvpn-linux/issues/424#issuecomment-2247172101

For others issues if the problem still reproduces please open separate issues, to better keep track of them.
Thanks
